src/hooks/useSearchAddressBook.ts

```diff
@@ -3,16 +3,24 @@ import {ethers} from "ethers";
 import {networks} from "../redux/networks";
 import {useAppSelector} from "../redux/hooks";
 import {addressBookSelectors} from "../redux/slices/addressBook.slice";
+import {ensApi} from "../redux/slices/ensResolver.slice";
 
 export const useSearchAddressBook = (searchTerm: string) => {
   const isSearchTermAddress = useMemo(
     () => ethers.utils.isAddress(searchTerm),
     [searchTerm]
   );
 
+
+  const ensQuery = ensApi.useResolveNameQuery(
+    searchTerm
+  )
+
+  const ensSearchTerm = ensQuery !== null ? ensQuery.address : searchTerm.toLocaleLowerCase()
```

Comment:
I'm a bit confused -- shouldn't `ensQuery.address` be `ensQuery.data?.address`? If so, please run `yarn build` before pushing any code because this project uses TypeScript and it would catch such type errors straight away. An IDE like Visual Studio Code would catch this type of errors straight away as well.